client/app.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import axios from 'axios';
+import SellerInfo from './sellerInfo.js';
+import ItemName from './itemName.js';
+import ItemPrice from './itemPrice.js';
+import FreeShipping from './freeShipping.js';
+import OnOrderAvailable from './onOrderAvailable.js';
+import Badge from './badge.js';
+
+class App extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      sellerName: '',
+      averageReviewScore: null,
+      numberReviews: null,
+      itemName: '',
+      badge: '',
+      itemPrice: null,
+      freeShipping: false,
+      productOptions: [],
+      personalization: false,
+      availableQuantity: null,
+      onOrder: null,
+      productId: null
+    };
+    this.requestProductDetails = this.requestProductDetails.bind(this);
+    this.updateState = this.updateState.bind(this);
+  }
+
+  async requestProductDetails(productId) {
+    try {
+      var response = await axios.get(`http://127.0.0.1:1234/api/checkout/${productId}/details`);
+    } catch (err) {
+      console.log(err);
+    }
+    return response.data;
+  }
+
+  updateState(newData) {

```

Comment:
```js
this.setState({
    ...this.state,
    ...newData
});
```
I have never used Object.assign to change the state but you can try doing it like this so it looks cleaner.

Reply:
I was doing this because my state has alot of props and  not necessarily all of them will be changed on each update so it's easier than writing them individually

Reply:
Ah nevermind I see what you're saying. react has some convenience built in that automatically does this for you.